cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/PullCommand.java

```diff
@@ -0,0 +1,154 @@
+/*
+ *  Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package io.ballerina.cli.cmd;
+
+import io.ballerina.projects.utils.ProjectConstants;
+import org.ballerinalang.central.client.CentralAPIClient;
+import org.ballerinalang.tool.BLauncherCmd;
+import org.ballerinalang.tool.LauncherUtils;
+import org.wso2.ballerinalang.compiler.util.Names;
+import org.wso2.ballerinalang.util.RepoUtils;
+import picocli.CommandLine;
+
+import java.io.IOException;
+import java.io.PrintStream;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.regex.Pattern;
+
+import static io.ballerina.cli.cmd.Constants.PULL_COMMAND;
+import static io.ballerina.runtime.util.RuntimeConstants.SYSTEM_PROP_BAL_DEBUG;
+import static java.nio.file.Files.createDirectory;
+import static org.ballerinalang.tool.LauncherUtils.createLauncherException;
+import static org.wso2.ballerinalang.programfile.ProgramFileConstants.SUPPORTED_PLATFORMS;
+
+/**
+ * This class represents the "ballerina pull" command.
+ *
+ * @since 2.0.0
+ */
+@CommandLine.Command(name = PULL_COMMAND,
+        description = "download the module source and binaries from a remote repository")
+public class PullCommand implements BLauncherCmd {
+    //module name format : <org-name>/<package-name> | <org-name>/<package-name>:<version>
+    //version format : 1, 1.*, 1.*.*
+    private static final String PKG_NAME_REGEX = "[^0-9_][_\\w]+/[^0-9_][_\\.\\w]+|" +
```

Comment:
Probably we should move this validation to project api

Reply:
added to `ProjectConstants` class